server/sonar-server/src/main/java/org/sonar/ce/queue/report/ReportSubmitter.java

```diff
@@ -45,35 +49,53 @@
   private final ReportFiles reportFiles;
   private final ComponentService componentService;
   private final PermissionService permissionService;
+  private final DbClient dbClient;
 
   public ReportSubmitter(CeQueue queue, UserSession userSession, ReportFiles reportFiles,
-    ComponentService componentService, PermissionService permissionService) {
+    ComponentService componentService, PermissionService permissionService, DbClient dbClient) {
     this.queue = queue;
     this.userSession = userSession;
     this.reportFiles = reportFiles;
     this.componentService = componentService;
     this.permissionService = permissionService;
+    this.dbClient = dbClient;
   }
 
   public CeTask submit(String projectKey, @Nullable String projectBranch, @Nullable String projectName, InputStream reportInput) {
     String effectiveProjectKey = ComponentKeys.createKey(projectKey, projectBranch);
     ComponentDto project = componentService.getNullableByKey(effectiveProjectKey);
     if (project == null) {
-      // the project does not exist -> require global permission
-      userSession.checkPermission(SCAN_EXECUTION);
+      project = createProject(projectKey, projectBranch, projectName);
+    }
+
+    userSession.checkComponentPermission(SCAN_EXECUTION, projectKey);
```

Comment:
Hum, if the user has not the scan permission on the project, the project will be created, no ?